### PR TITLE
Use an element prop for setting editor and drag handle

### PR DESCRIPTION
**Pull request by amanmahajan7**

### CODE REVIEW

File: src/DataGrid.tsx [hunk 1]
```diff
@@ -940,32 +950,40 @@ function DataGrid<R, SR, K extends Key>(
     return isDraggedOver ? selectedPosition.idx : undefined;
   }
 
-  function getSelectedCellProps(rowIdx: number): SelectedCellProps | EditCellProps<R> | undefined {
-    if (selectedPosition.rowIdx !== rowIdx) return;
+  function getSelectedCellProps(rowIdx: number): SelectedCellProps<R, SR> | undefined {
```
Comment (nstepien) [inline src/DataGrid.tsx:953]: Could destructure `selectedPosition` so we don't repeat property access on `idx` and `rowIdx`

File: src/DataGrid.tsx [hunk 2]
```diff
@@ -940,32 +950,40 @@ function DataGrid<R, SR, K extends Key>(
...
+        onFocus: handleFocus,
         onKeyDown: handleKeyDown,
-        editorProps: {
-          editorPortalTarget,
-          row: selectedPosition.row,
-          onRowChange: handleEditorRowChange,
-          onClose: handleOnClose
-        }
+        selectedCellEditor: (
+          <EditCell
```
Comment (nstepien) [inline src/DataGrid.tsx:967]: It's pushed to an array so we still need a `key` prop here.
Check the jest output.
```suggestion
          <EditCell
            key={column.key}
```

File: src/DataGrid.tsx [hunk 3]
```diff
@@ -940,32 +950,41 @@ function DataGrid<R, SR, K extends Key>(
...
 
-  function getSelectedCellProps(rowIdx: number): SelectedCellProps | EditCellProps<R> | undefined {
+  function getSelectedCellProps(rowIdx: number): SelectedCellProps<R, SR> | undefined {
     if (selectedPosition.rowIdx !== rowIdx) return;
 
     if (selectedPosition.mode === 'EDIT') {
+      const row = rows[selectedPosition.rowIdx];
+      if (isGroupRow(row)) return;
+      const column = columns[selectedPosition.idx];
+      const colSpan = getColSpan(column, lastFrozenColumnIndex, { type: 'ROW', row });
```
Comment (amanmahajan7) [inline src/DataGrid.tsx:960]: May be we can use `row: selectedPosition.row`

File: src/types.ts [hunk 4]
```diff
@@ -153,13 +133,15 @@ export interface RowRendererProps<TRow, TSummaryRow = unknown>
   viewportColumns: readonly CalculatedColumn<TRow, TSummaryRow>[];
   row: TRow;
   rowIdx: number;
+  selectedCellIdx?: number;
```
Comment (nstepien) [inline src/types.ts:136]:
```suggestion
  selectedCellIdx: number | undefined;
```

File: src/types.ts [hunk 5]
```diff
@@ -153,13 +133,15 @@ export interface RowRendererProps<TRow, TSummaryRow = unknown>
...
   copiedCellIdx: number | undefined;
   draggedOverCellIdx: number | undefined;
   lastFrozenColumnIndex: number;
   isRowSelected: boolean;
   top: number;
   height: number;
-  selectedCellProps: EditCellProps<TRow> | SelectedCellProps | undefined;
+  selectedCellEditor?: ReactElement<EditorProps<TRow>>;
+  selectedCellDragHandle?: ReactElement<React.HTMLAttributes<HTMLDivElement>>;
```
Comment (nstepien) [inline src/types.ts:144]:
```suggestion
  selectedCellEditor: ReactElement<EditorProps<TRow>> | undefined;
  selectedCellDragHandle: ReactElement<React.HTMLAttributes<HTMLDivElement>> | undefined;
```

**Review by nstepien — CHANGES_REQUESTED** [review]

File: src/DataGrid.tsx (hunk 3 above)
Comment (nstepien) [reply src/DataGrid.tsx:960]: Worth a try

File: src/Row.tsx [hunk 6]
```diff
@@ -94,9 +79,9 @@ function Row<R, SR>(
           isCopied={copiedCellIdx === idx}
           isDraggedOver={draggedOverCellIdx === idx}
           isCellSelected={isCellSelected}
-          dragHandleProps={dragHandleProps}
-          onFocus={onFocus}
-          onKeyDown={onKeyDown}
+          dragHandle={isCellSelected ? selectedCellDragHandle : undefined}
+          onFocus={isCellSelected ? onFocus : undefined}
+          onKeyDown={isCellSelected ? onKeyDown : undefined}
```
Comment (nstepien) [inline src/Row.tsx:84]: What if we pass `onFocus`/`onKeyDown` as is all the time? 🤔

Comment (amanmahajan7) [reply src/Row.tsx:84]: It would rerender all the cells then

File: src/types.ts (hunk 5 above)
Comment (amanmahajan7) [reply src/types.ts]: Ah, I remember why I used `?:` instead. `Row` complains as `getSelectedCellProps` also returns undefined

**Review by nstepien — APPROVED** [review]

**Pull request merged**